source/CCustomOpcodeSystem.cpp

```diff
@@ -1,2697 +1,2794 @@
+	{
+		TRACE("Injecting CustomOpcodeSystem...");
+		CGameVersionManager& gvm = GetInstance().VersionManager;
+		oldOpcodeHandlerTable = gvm.TranslateMemoryAddress(MA_OPCODE_HANDLER);
+
+		// add handler for custom opcodes
+		oldOpcodeHandlerTable[27] = reinterpret_cast<_OpcodeHandler>(customOpcodeHandler);
+
+		// replace old OpcodeHandlerTable with the new one
+		//inj.MemoryWrite(gvm.TranslateMemoryAddress(MA_OPCODE_HANDLER_REF), reinterpret_cast<_OpcodeHandler>(&newOpcodeHandlerTable[0]));
+		MemWrite(gvm.TranslateMemoryAddress(MA_OPCODE_HANDLER_REF), reinterpret_cast<_OpcodeHandler>(&newOpcodeHandlerTable[0]));
+
+		// copy old table to the new                                                             
+		//std::copy(oldOpcodeHandlerTable, oldOpcodeHandlerTable + 28, newOpcodeHandlerTable);
+		MemCopy<_OpcodeHandler>(newOpcodeHandlerTable, oldOpcodeHandlerTable, (&oldOpcodeHandlerTable[28] - oldOpcodeHandlerTable) * 4);
+		//MemCopy(newOpcodeHandlerTable, oldOpcodeHandlerTable, oldOpcodeHandlerTable - (oldOpcodeHandlerTable + 28));
+
+		// fill the rest with default handler
+		std::fill(newOpcodeHandlerTable + 28, newOpcodeHandlerTable + 329, reinterpret_cast<_OpcodeHandler>(extraOpcodeHandler));
+
+		FUNC_fopen = gvm.TranslateMemoryAddress(MA_FOPEN_FUNCTION);
+		FUNC_fclose = gvm.TranslateMemoryAddress(MA_FCLOSE_FUNCTION);
+		FUNC_fread = gvm.TranslateMemoryAddress(MA_FREAD_FUNCTION);
+		FUNC_fwrite = gvm.TranslateMemoryAddress(MA_FWRITE_FUNCTION);
+		FUNC_fgetc = gvm.TranslateMemoryAddress(MA_FGETC_FUNCTION);
+		FUNC_fgets = gvm.TranslateMemoryAddress(MA_FGETS_FUNCTION);
+		FUNC_fputs = gvm.TranslateMemoryAddress(MA_FPUTS_FUNCTION);
+		FUNC_fseek = gvm.TranslateMemoryAddress(MA_FSEEK_FUNCTION);
+		FUNC_fprintf = gvm.TranslateMemoryAddress(MA_FPRINTF_FUNCTION);
+		FUNC_ftell = gvm.TranslateMemoryAddress(MA_FTELL_FUNCTION);
+		FUNC_fflush = gvm.TranslateMemoryAddress(MA_FFLUSH_FUNCTION);
+		FUNC_feof = gvm.TranslateMemoryAddress(MA_FEOF_FUNCTION);
+		FUNC_ferror = gvm.TranslateMemoryAddress(MA_FERROR_FUNCTION);
+
+		pedPool = gvm.TranslateMemoryAddress(MA_PED_POOL);
+		vehiclePool = gvm.TranslateMemoryAddress(MA_VEHICLE_POOL);
+		objectPool = gvm.TranslateMemoryAddress(MA_OBJECT_POOL);
+		GetUserDirectory = gvm.TranslateMemoryAddress(MA_GET_USER_DIR_FUNCTION);
+		ChangeToUserDir = gvm.TranslateMemoryAddress(MA_CHANGE_TO_USER_DIR_FUNCTION);
+		ChangeToProgramDir = gvm.TranslateMemoryAddress(MA_CHANGE_TO_PROGRAM_DIR_FUNCTION);
+		FindGroundZ = gvm.TranslateMemoryAddress(MA_FIND_GROUND_Z_FUNCTION);
+		GetPlayerPed = gvm.TranslateMemoryAddress(MA_GET_PLAYER_PED_FUNCTION);
+		Handling = gvm.TranslateMemoryAddress(MA_HANDLING);
+		Models = gvm.TranslateMemoryAddress(MA_MODELS);
+		SpawnCar = gvm.TranslateMemoryAddress(MA_SPAWN_CAR_FUNCTION);
+
+		if (gvm.GetGameVersion() == GV_US10) {
+			// make it compatible with fastman92's limit adjuster
+			RadarBlips = injector::ReadMemory<CMarker*>(0x583A05 + 2, true);

```

Comment:
Will this still work if f92la is not installed?

Reply:
Yes, I just read the address where the original pointer is stored, f92la only changes the pointer.

Btw f92la is 1.0US only (and it will probably be forever), so therefore it is not necessary to do the same with any other version.

Reply:
I would prefer to write version-agnostic code (e.g. if we read a pointer in v1.0 do the same in other versions too, so you don't have different behavior on different versions), but this is up to you.

Reply:
The code would be more pleasant, but in my mods I never worry about different versions, so I don't even have all the IDBs that CLEO supports, it would be time spent on something that, in practice, in this specific case, isn't even required.
But I understand your point, this is a pattern break. Well, let someone else do this.

Reply:
can you please leave a todo comment near this change, so we can keep track of it? unfortunately due to formatting github does not display the changes made in this file.

Reply:
Added. Also for CCustomScript::CCustomScript

Reply:
can't see a todo for CCustomScript::CCustomScript, did you forget to commit?

Reply:
Ah, I misread a file name, sorry.

Reply:
alright, thanks 👍